chainer/functions/noise/dropout.py

```diff
@@ -33,13 +37,33 @@ def forward(self, x):
             y = x[0] * self.mask
         else:
             scale = x[0].dtype.type(1. / (1 - self.dropout_ratio))
-            xp = cuda.get_array_module(*x)
-            if xp == numpy:
-                flag = xp.random.rand(*x[0].shape) >= self.dropout_ratio
-                self.mask = scale * flag
+            flag = numpy.random.rand(*x[0].shape) >= self.dropout_ratio
+            self.mask = scale * flag
+            y = x[0] * self.mask
+        return y,
+
+    def forward_gpu(self, x):
+        if (chainer.should_use_cudnn('==always', 5000) and
+                x[0].flags.c_contiguous):
```

Comment:
Should I change it to:
```python
if (chainer.should_use_cudnn('==always', 5000) and x[0].flags.c_contiguous and self.mask is None):
```

It seems that now dropout function has `mask` argument in its constructor for making results deterministic, but cuDNN does not support `mask` (same as iDeep).